src/main/java/hudson/plugins/s3/S3Profile.java

```diff
@@ -203,30 +198,26 @@ private <T> T invoke(boolean uploadFromSlave, FilePath filePath, MasterSlaveCall
     }
 
     public List<String> list(Run build, String bucket) {
-        final AmazonS3 s3client = getClient(ClientHelper.DEFAULT_AMAZON_S3_REGION_NAME);
+        final S3Client s3client = getClient(ClientHelper.DEFAULT_AMAZON_S3_REGION_NAME);
 
         final String buildName = build.getDisplayName();
         final int buildID = build.getNumber();
         final Destination dest = new Destination(bucket, "jobs/" + buildName + '/' + buildID + '/' + name);
 
-        final ListObjectsRequest listObjectsRequest = new ListObjectsRequest()
-        .withBucketName(dest.bucketName)
-        .withPrefix(dest.objectName)
-        .withEncodingType("url");
+        final ListObjectsRequest listObjectsRequest = ListObjectsRequest.builder()
```

Comment:
why not using ListObjectsV2Request?
can you have more than 1000 objects returned? 
if yes you should look at continuation token.

Reply:
fixed